calicoctl/calico_ctl/node.py

```diff
@@ -258,9 +258,12 @@ def node_start(node_image, runtime, log_dir, ip, ip6, as_num, detach,
         if not etcd_ok or (using_docker and not docker_ok):
             sys.exit(1)
 
-    # Ensure log directory exists
+    # Ensure log directory and /var/run/calico exist so that they can be
+    # mounted into the containers.
     if not os.path.exists(log_dir):
         os.makedirs(log_dir)
+    if not os.path.exists("/var/run/calico"):
+        os.makedirs("/var/run/calico")
```

Comment:
It's better (no race condition) and more Pythonic (forgiveness vs permission) to catch an exception here (although the verbosity does make me sad).

Sadly the exist_ok parameter isn't available until 3.2.